# HG changeset patch
# User Jose Luis Rivero <jrivero@osrfoundation.org>
# Date 1406833352 -7200
# Branch sdf_2.0_cmake_config
# Node ID 91455b9e17e4d00ce196634876ec3c80843d63b3
# Parent  f24911daff19eb38606b41f7e564b4199c8e921c
Implement cmake version file

diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.8 FATAL_ERROR)
+cmake_minimum_required(VERSION 2.8.6 FATAL_ERROR)
 
 if(COMMAND CMAKE_POLICY)
   CMAKE_POLICY(SET CMP0003 NEW)
@@ -224,8 +224,23 @@
   set(PKG_LIBRARIES sdformat)
   set(PKG_DEPENDS Boost)
   set(cmake_conf_file "cmake/sdf_config.cmake")
+  set(cmake_conf_version_file "cmake/SDFormatConfigVersion.cmake")
   configure_file("${CMAKE_CURRENT_SOURCE_DIR}/${cmake_conf_file}.in" "${CMAKE_CURRENT_BINARY_DIR}/SDFormatConfig.cmake" @ONLY)
-  install(FILES ${CMAKE_CURRENT_BINARY_DIR}/SDFormatConfig.cmake DESTINATION ${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME_LOWER}/ COMPONENT cmake)
+  # Use write_basic_package_version_file to generate a ConfigVersion file that
+  # allow users of gazebo to specify the API or version to depend on
+  # TODO: keep this instruction until deprecate Ubuntu/Precise and update with
+  # https://github.com/Kitware/CMake/blob/v2.8.8/Modules/CMakePackageConfigHelpers.cmake 
+  include(WriteBasicConfigVersionFile)
+  write_basic_config_version_file(
+      ${CMAKE_CURRENT_BINARY_DIR}/${cmake_conf_version_file}
+      VERSION "${SDF_VERSION_FULL}"
+      COMPATIBILITY SameMajorVersion)
+  install(FILES 
+            ${CMAKE_CURRENT_BINARY_DIR}/SDFormatConfig.cmake
+            ${CMAKE_CURRENT_BINARY_DIR}/${cmake_conf_version_file}
+          DESTINATION 
+	    ${LIB_INSTALL_DIR}/cmake/${PROJECT_NAME_LOWER}/ 
+	  COMPONENT cmake)
 
   ########################################
   # Package Creation: 
